ops/jenkins/bfd-deploy-apps/Jenkinsfile

```
@@ -0,0 +1,338 @@
+                    gitCommitId = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+
+                    // Get the remote repo url. This assumes we are using git+https not git+ssh.
+                    gitRepoUrl = sh(returnStdout: true, script: 'git config --get remote.origin.url').trim().replaceAll(/\.git$/, '')
+
+                    // Load the child Jenkinsfiles.
+                    migratorScripts = load('ops/terraform/services/migrator/deploy.groovy')
+                    serverScripts = load('ops/terraform/services/server/deploy.groovy')
+
+                    trimmedEnv = params.env.trim()
+                    lockResource = trimmedEnv == 'prod-sbx' ? 'env_prod_sbx' : "env_${trimmedEnv}"
+
+                    awsAuth.assumeRole()
+                }
+            }
+        }
+
+        stage('Set Branch Name') {
+            steps {
+                script {
+                    currentStage = env.STAGE_NAME
+                    if (env.BRANCH_NAME.startsWith('PR')) {
+                        gitBranchName = env.CHANGE_BRANCH
+                    } else {
+                        gitBranchName = env.BRANCH_NAME
+                    }
+                }
+            }
+        }
+
+        stage('Deploy Base') {
+            steps {
+                script {
+                    currentStage = env.STAGE_NAME
+                    try {
+                        lock(resource: lockResource) {
+                            awsAuth.assumeRole()
+                            terraform.deployTerraservice(
+                                    env: trimmedEnv,
+                                    directory: 'ops/terraform/services/base'
+                            )
+                        }
+                    } catch (org.jenkinsci.plugins.workflow.steps.FlowInterruptedException e) {
+                        currentBuild.result = 'ABORTED'
+                        sendNotifications(trimmedEnv, currentBuild.currentResult, currentStage, gitCommitId, gitRepoUrl)
+                        throw e
+                    } catch (ex) {
+                        currentBuild.result = 'FAILURE'
+                        sendNotifications(trimmedEnv, currentBuild.currentResult, currentStage, gitCommitId, gitRepoUrl)
+                        throw ex
+                    }
+                }
+            }
+        }
+
+        stage('Deploy Migrator') {
+            steps {
+                script {
+                    currentStage = env.STAGE_NAME
+                    try {
+                        def mgiratorAmiOverride = (params.migrator_ami_override?.trim())
+                                                        ? params.migrator_ami_override.trim()
+                                                        : ""
+
+                        lock(resource: lockResource) {
+                            migratorDeploymentSuccessful = migratorScripts.deployMigrator(
+                                amiId: mgiratorAmiOverride,
```

```suggestion
                        def migratorAmiOverride = (params.migrator_ami_override?.trim())
                                                        ? params.migrator_ami_override.trim()
                                                        : ""

                        lock(resource: lockResource) {
                            migratorDeploymentSuccessful = migratorScripts.deployMigrator(
                                amiId: migratorAmiOverride,
```

Aside from the typo... I'm a little surprised by this. I haven't given this enough thought, but maybe you can help–how does this behave when `deployMigrator()` receives an empty string for the ami override?

---

Decided to revert this as it seems a bit excessive.